bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/TableOfContentsFilter.java

```diff
@@ -118,48 +123,53 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             return;
         }
 
+        if (!Optional.ofNullable((Boolean) request.getAttribute(TableOfContentsImpl.TOC_REQUEST_ATTR_FLAG)).orElse(Boolean.FALSE)) {
+            LOGGER.debug("{} component not present on page, so not parsing the page output",
+                TableOfContents.class.getName());
+            chain.doFilter(request, response);
+            return;
+        }
+
+        if (!Optional.ofNullable(response.getContentType()).filter(ct -> ct.contains("text/html")).isPresent()) {
+            LOGGER.debug("Response content type not \"text/html\", bypassing filter");
+            chain.doFilter(request, response);
+            return;
+        }
+
```

Comment:
I tried it out on a sample page but the TOC renders empty. I believe it's because the request attribute is set later in the chain by the TOC component. Following code would work:
```
        if (!enabled) {
            LOGGER.debug("{} not enabled, bypassing it", TableOfContentsFilter.class.getName());
            chain.doFilter(request, response);
            return;
        }

        CharResponseWrapper responseWrapper = new CharResponseWrapper((HttpServletResponse) response);
        chain.doFilter(request, responseWrapper);
        String originalContent = responseWrapper.toString();

        if (!Optional.ofNullable(response.getContentType()).filter(ct -> ct.contains("text/html")).isPresent()) {
            LOGGER.debug("Response content type not \"text/html\", bypassing filter");
            response.getWriter().write(originalContent);
            return;
        }

        Boolean containsTableOfContents = (Boolean) request.getAttribute(TableOfContentsImpl.TOC_REQUEST_ATTR_FLAG);
        if (containsTableOfContents == null || !containsTableOfContents) {
            LOGGER.debug("request attribute {} not present or set to false, bypassing the filter {}", TableOfContentsImpl.TOC_REQUEST_ATTR_FLAG, TableOfContentsFilter.class.getName());
            response.getWriter().write(originalContent);
            return;
        }

        Document document = Jsoup.parse(originalContent);
```